File: guides/micronaut-microstream-persistence/kotlin/src/main/kotlin/example/micronaut/Fruit.kt
package example.micronaut

import io.micronaut.serde.annotation.Serdeable

import jakarta.validation.constraints.NotBlank

@Serdeable // <1>
data class Fruit(
    @field:NotBlank val name: String, // <2>
    var description: String? // <3>
)

File: guides/micronaut-data-mongodb-asynchronous/python/src/example/micronaut/fruit.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank
from micronaut.data.annotation import GeneratedValue, Id, MappedEntity
from micronaut.serde.annotation import Serdeable


@Serdeable
@MappedEntity  # <1>
@dataclass
class Fruit:
    name: Annotated[str, NotBlank]  # <3>
    description: str | None = None
    id: Annotated[str | None, Id, GeneratedValue] = None  # <2>
